polygeist/tools/cgeist/Lib/CGExpr.cc

```diff
@@ -2368,6 +2366,83 @@ ValueCategory MLIRScanner::EmitScalarConversion(ValueCategory Src,
   return Res;
 }
 
+ValueCategory MLIRScanner::EmitFloatToBoolConversion(ValueCategory Src) {
+  assert(Src.val.getType().isa<FloatType>() && "Expecting a float value");
+  mlir::OpBuilder subbuilder(builder.getContext());
+  subbuilder.setInsertionPointToStart(entryBlock);
+  auto FloatTy = cast<FloatType>(Src.val.getType());
+  auto Zero = subbuilder.create<ConstantFloatOp>(
+      builder.getUnknownLoc(),
```

Comment:
Would be better if we could use a source location that is not unknown, perhaps is possible to retrieve a source location that is more meaningful. Perhaps from the MLIRScanner.

Reply:
That's true... I can open a new PR addressing this in all points in which we're using `UnknownLoc` (there are plenty of them).

Reply:
Ok I'm fine with separate PR.